### ISSUE TITLE
Fully specified monorepos should do an exact match

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#14559]
### How are you running Renovate?

WhiteSource Renovate hosted app on github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

`react-native` is grouped with `react` packages in "react monorepo" mistakenly. This is because `https://github.com/facebook/react` is a prefix of `https://github.com/facebook/react-native`.

Reproduction: https://github.com/renovate-tests/npm96

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste any log here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

**Comment by rarkins** [comment]
I think we need a new match, which is just `matchSourceUrl` exactly, and monorepos should be in that.

**Comment by astellingwerf** [comment]
Wouldn't adding a slash at the end of the URL solve this?

**Comment by rarkins** [comment]
Only if we also add a trailing slash to our "sourceUrl" string before comparing

**Comment by MaronHatoum** [comment]
I believe we can remove `status:blocked`  since https://github.com/renovatebot/renovate/issues/14561 closed

**Comment by ikesyo** [comment]
Looks like this has been resolved by #18973.
